### PR TITLE
Add BackstageTab to overridableComponents

### PR DESCRIPTION
**Pull request by saan800**
Signed-off-by: sandra-greenhalgh <sandra.greenhalgh@justeattakeaway.com>

## Hey, I just made a Pull Request!

Add `BackstageTab` to `overridableComponents` so its possible to override styles in themes

Its a follow on from https://github.com/backstage/backstage/pull/10554

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by saan800** [comment]
Hi Backstage reviewer. How do we update the `api-report.md` ? Thanks

File: packages/core-components/src/components/Tabs/Tab.tsx [hunk 1]
```diff
@@ -37,6 +37,8 @@ const tabMarginLeft = (isFirstNav: boolean, isFirstIndex: boolean) => {
   return '40px';
 };
 
+export type TabClassKey = 'root' | 'selected';
```
Comment (benjdlambert) [inline packages/core-components/src/components/Tabs/Tab.tsx:40]: Think you'll need a `/** @public */` here to get rid of the warning in the api-report.md

**Review by benjdlambert — COMMENTED** [review]
Nice 👍 just one small nit!

**Review by benjdlambert — APPROVED** [review]
Good to ship once 🟢 !

File: packages/core-components/src/components/Tabs/Tab.tsx (hunk 1 above)
Comment (saan800) [reply packages/core-components/src/components/Tabs/Tab.tsx:41]: done

**Pull request merged**